### ISSUE TITLE
Feature Request: automatically refresh diagnostic plugin

### ISSUE THREAD

**Issue by HikeAndMap** [OpenDroneMap/WebODM#1453]
The html gui doesn't automatically update.. so I keep pressing ctrl-F5 to update the RAM consumption but also other fields ..

Is there a possibility to include an button on the upper right or maybe in the settings, if checked, then the GUI updates automatically every 5 seconds or so?

**Comment by MarkAtGit** [comment]
Maybe it's also nice to add the possibility to download some (basic) Diagnostic usage statistics from the last 12 hours, updated every minute (720 records) in this Diagnostic Plugin. The same way as you can now see the log file while processing projects.
![WebODM diagnostical info](https://github.com/user-attachments/assets/fbc9959f-82a4-4e84-a1e8-e8600186d9b0)

Best regards,

Mark 
Aeret

**Comment by pierotofy** [comment]
I'd welcome pull requests to improve the diagnostic plugin 🙏 if there's interest, feel free to open one?

**Comment by jet86** [comment]
For what it's worth, I had started some work on improving the Diagnostic Plugin a while ago, but got busy with other things - you can see what I'd started at https://github.com/jet86/WebODM/tree/diagnostics if interested.

TL;DR: switching to psutil for details on storage, memory, and swap so that it can be more consistent across different OS installs (Mac/Win/Linux), as well as adding CPU metrics.

It's not at a point that it's ready for a Pull Request (or even for me to take it to the forums to see if it was worthy of a PR) but I'll try to get back into working on it now that I have more time again.

PS: apologies if this comment reopens the feature request, just close it again if so.

**Comment by HikeAndMap** [comment]
I have to say, @jet86 I'll keep your development under observation.. I'm very happy to see what you're doing there
